dts/arm/silabs/efr32xg23.dtsi

```diff
@@ -0,0 +1,469 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+#include <dt-bindings/adc/adc.h>
+#include <dt-bindings/clock/silabs/xg23-clock.h>
+#include <freq.h>
+
+/ {
+	chosen {
+		zephyr,flash-controller = &msc;
+		zephyr,entropy = &se;
+	};
+
+	clocks {
+		hfxort: hfxort {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hfxo>;
+		};
+		hfrcodpllrt: hfrcodpllrt {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hfrcodpll>;
+		};
+		sysclk: sysclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hfrcodpll>;
+		};
+		hclk: hclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&sysclk>;
+			/* Divider 1, 2, 4, 8, or 16 */
+			clock-div = <1>;
+		};
+		pclk: pclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hclk>;
+			/* Divider 1 or 2 */
+			clock-div = <2>;
+		};
+		lspclk: lspclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&pclk>;
+			/* Fixed divider of 2 */
+			clock-div = <2>;
+		};
+		hclkdiv1024: hclkdiv1024 {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hclk>;
+			/* Fixed divider of 1024 */
+			clock-div = <1024>;
+		};
+		traceclk: traceclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&sysclk>;
+			/* Divider 1, 2, 3 or 4 */
+			clock-div = <1>;
+		};
+		em01grpaclk: em01grpaclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hfrcodpll>;
+		};
+		em01grpcclk: em01grpcclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hfrcodpll>;
+		};
+		iadcclk: iadcclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&em01grpaclk>;
+		};
+		em23grpaclk: em23grpaclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&lfrco>;
+		};
+		em4grpaclk: em4grpaclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&lfrco>;
+		};
+		sysrtcclk: sysrtcclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&lfrco>;
+		};
+		wdog0clk: wdog0clk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&lfrco>;
+		};
+		wdog1clk: wdog1clk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&lfrco>;
+		};
+		pcnt0clk: pcnt0clk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&em23grpaclk>;
+		};
+		eusart0clk: eusart0clk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&em01grpcclk>;
+		};
+		systickclk: systickclk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&hclk>;
+		};
+		vdac0clk: vdac0clk {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+			clocks = <&em01grpaclk>;
+		};
```

Comment:
Where is there a full list of all the clocks that should be defined?

Reply:
Either refer to the reference manual (CMU_<>CLKCTRL registers) or the clock manager config header in the Simplicity SDK: https://github.com/SiliconLabs/simplicity_sdk/blob/19a4476b12e9e827cbdd89c58141494fea188a67/platform/service/clock_manager/config/EFR32ZG23/sl_clock_manager_tree_config.h

From what I could tell, the LCDCLK is the only node missing.

Reply:
In the RM I see there is a "CMU_EXPORTCLKCTRL" for example that doesn't exist in the clock manager config header. What does this mean? Should it be in the DTS?

Reply:
EXPORTCLK is interesting, since it's a clock tree node that is only available as an input to the CLKOUT feature. Since it's not used by the clock manager config header, adding it in DTS doesn't achieve anything other than modeling the hardware more completely. If it were added, it would be a node `exportclk` with source `sysclk` and a divider.

When it comes to sorting, IMO we can postpone that until we have scripts to fully generate DTS. At that point, the order will be whatever the script does, which should be some variant of order of appearance, such that parent nodes in the clock tree are defined before their children that depend on them.

Reply:
Ok, so in the meantime I won't add `exportclk`. What about `lesensehfclk`

Reply:
Also, is this automation/scripts in the works? Is there a PR to look at?

Reply:
You're right, I missed `lesensehfclk`. It should be added similarly to `lcdclk`, as the clock manager does use it. Its default source clock should be `fsrco`.

As for the automation, it's unfortunately not started yet. We hope to get to it early next year, as it would make it way easier to maintain the DTS content across Series 2, and make it possible to support all the OPNs within each device family, not just the individual OPNs needed to enable certain boards.